### ISSUE TITLE
Response->cache() broken on Apache 2.4+ with mod_deflate

### ISSUE THREAD

**Issue by bertptrs** [mako-framework/framework#163]
When using `mod_deflate` to add gzip compression to text resources, apache append any `ETag` present with the suffix `-gzip`.

`$response->cache()` checks whether the header sent matches the header recieved, but recieves the appended version (as apache does not un-append the request again). This results in being reckognized as a cache-miss and sending the request again.

An apache-based workaround can be found in their [their documentation](https://httpd.apache.org/docs/trunk/mod/mod_deflate.html#deflatealteretag) disabling the `AlterETag` option. This however is not appropriate behavior, as 

> This was the default prior to 2.4.0, but does not satisfy the HTTP/1.1 property that all representations of the same resource have unique ETags.

A solution would be to check only the first `n` characters of the ETag response.

Mako version: 4.5.7
Apache version: 2.4.7
PHP version: 5.5.9

**Comment by freost** [comment]
Hi,

I haven't had time to set up a VM to test this yet but does it append the suffix after the double quotes or between them?

```
"c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2-gzip"
```

vs

```
"c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2"-gzip
```

**Comment by bertptrs** [comment]
It adds it between the quotes. For example: `"555d213a9e976e259c9db80dc465359dcf8998ebf3d2c39c7aaab44b838e76d1-gzip"`.

By the way, I'm running Ubuntu 14.04 server fully updated.

**Comment by freost** [comment]
Fixed in 4.5.9 :)

**Comment by bertptrs** [comment]
Thanks heaps, as always.
